### ISSUE TITLE
Code generated with error

### ISSUE THREAD

**Issue by wborowiecki** [Avanade/NTangle#25]
Generated code: 

                // Join table: '[dbo].[NAC_RegionDetails]' (unique name 'Region')
                new MultiSetCollArgs<VendorsCdc.RegionCdcCollection, VendorsCdc.RegionCdc>(_regionCdcMapper, r =>
                {
                    foreach (var r in r.GroupBy(x => new { x.RDRegionID }).Select(g => new { g.Key.RDRegionID, Coll = g.ToCollection<VendorsCdc.RegionCdcCollection, VendorsCdc.RegionCdc>() }))
                    {
                        nvColl.Where(x => x.VERegionID == r.RDRegionID).ForEach(x => x.Region = r.Coll.FirstOrDefault());
                    }
                }),

error: 

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0136	A local or parameter named 'r' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter	FMSCDC.Publisher	VendorsCdcOrchestrator.cs	68	Active

In short "r" is used twice, once in "r =>" and then in the inner loop "foreach (var r in r.".
